src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.Ssl.cs

```diff
@@ -3,12 +3,113 @@
 
 using System;
 using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+
 using Microsoft.Win32.SafeHandles;
 
+using SafeSslHandle = System.Net.SafeSslHandle;
+
 internal static partial class Interop
 {
     internal static partial class AndroidCrypto
     {
+        internal unsafe delegate PAL_SSLStreamStatus SSLReadCallback(byte* data, int* length);
+        internal unsafe delegate void SSLWriteCallback(byte* data, int length);
+
+        internal enum PAL_SSLStreamStatus
+        {
+            OK = 0,
+            NeedData = 1,
+            Error = 2,
+            Renegotiate = 3,
+            Closed = 4,
+        };
+
+        [DllImport(Interop.Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_SSLStreamCreate")]
+        internal static extern SafeSslHandle SSLStreamCreate();
+
+        [DllImport(Interop.Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_SSLStreamCreateWithCertificates")]
+        private static extern SafeSslHandle SSLStreamCreateWithCertificates(
+            ref byte pkcs8PrivateKey,
+            int pkcs8PrivateKeyLen,
+            PAL_KeyAlgorithm algorithm,
+            IntPtr[] certs,
+            int certsLen);
+        internal static SafeSslHandle SSLStreamCreateWithCertificates(ReadOnlySpan<byte> pkcs8PrivateKey, PAL_KeyAlgorithm algorithm, IntPtr[] certificates)
+        {
+            return SSLStreamCreateWithCertificates(
+                ref MemoryMarshal.GetReference(pkcs8PrivateKey),
+                pkcs8PrivateKey.Length,
+                algorithm,
+                certificates,
+                certificates.Length);
+        }
+
+        [DllImport(Interop.Libraries.CryptoNative, EntryPoint = "AndroidCryptoNative_SSLStreamInitialize")]
+        private static extern int SSLStreamInitializeImpl(
+            SafeSslHandle sslHandle,
+            [MarshalAs(UnmanagedType.U1)] bool isServer,
+            SSLReadCallback streamRead,
+            SSLWriteCallback streamWrite,
+            int appBufferSize);
+        internal static void SSLStreamInitialize(
+            SafeSslHandle sslHandle,
+            [MarshalAs(UnmanagedType.U1)] bool isServer,
```

Comment:
```suggestion
            bool isServer,
```

Since this isn't the P/Invoke the MarshalAs has no meaning (IIRC).